GVRf/Framework/framework/src/main/java/org/gearvrf/GVRContext.java

```diff
@@ -2679,4 +2685,19 @@ public Object getTag() {
         return mTag;
     }
 
+    private final HandlerThread mHandlerThread;
+    private final Handler mHandler;
+
+    public void runOnTheFrameworkThread(final Runnable runnable) {
```

Comment:
Looks like the framework thread is only designed as a queue. I think maybe its behavior and interaction with GL thread needs to be designed too?